internal/controller/suite_test.go

```diff
@@ -65,6 +64,10 @@ var dnsProviderFactory = &providerFake.Factory{
 	},
 }
 
+func SetDNSProvider(p provider.Provider) {
+	dnsProvider = p
+}
```

Comment:
I think we should avoid changing suite variables in tests. You won't be able to run tests in parallel if you do this as you will be modifying the provider that all specs will get back through the factory.

Reply:
Yeah just not really sure what other option we have

Reply:
We need a way to simulate errors from the provider IMO. Perhaps we create a follow up to add some form of error injection into the inmem provider?
Like if a test could set a response keyed to itself that would work